data-serving/data-service/src/model/location.ts

```diff
@@ -1,37 +1,32 @@
 import mongoose from 'mongoose';
 
-const fieldRequiredValidator = [
-    function (this: LocationDocument): boolean {
-        return (
-            this != null &&
-            this.country == null &&
-            this.administrativeAreaLevel1 == null &&
-            this.administrativeAreaLevel2 == null &&
-            this.administrativeAreaLevel3 == null
-        );
+const geometrySchema = new mongoose.Schema(
+    {
+        latitude: {
+            type: Number,
+            min: -90.0,
+            max: 90.0,
+            required: true,
+        },
+        longitude: {
+            type: Number,
+            min: -180.0,
+            max: 180.0,
+            required: true,
+        },
     },
-    'One of country, administrativeAreaLevel1, administrativeAreaLevel2, ' +
-        'or administrativeAreaLevel3 is required',
-];
+    { _id: false },
+);
 
 export const locationSchema = new mongoose.Schema(
     {
         country: {
             type: String,
-            required: fieldRequiredValidator,
-        },
-        administrativeAreaLevel1: {
-            type: String,
-            required: fieldRequiredValidator,
-        },
-        administrativeAreaLevel2: {
-            type: String,
-            required: fieldRequiredValidator,
-        },
-        administrativeAreaLevel3: {
-            type: String,
-            required: fieldRequiredValidator,
+            required: true,
         },
+        administrativeAreaLevel1: String,
+        administrativeAreaLevel2: String,
+        administrativeAreaLevel3: String,
         // Place represents a precise location, such as an establishment or POI.
         place: String,
         // A human-readable name of the location.
```

Comment:
Do we rely on `name` to the extent that it should be required? It'll be present in all geocoded results, IIRC, but I'm not sure if there's a use case to not have it.

Reply:
yeah, good call, it should be -- added it